### PR TITLE
feat: App group create update delete

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description
This PR includes the changes of app group create update and delete. Changed the logic of filtering from BE

Fixes # [AB#2676](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/2676)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Manually tested by creating updating multiple app groups

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ApplicationGroup/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    setSearchText,
+    searchApplied,
+    setSearchApplied,
+}: {
+    placeholder: string
+    handleFilterChanges: (searchText: string) => void
+    searchText: string
+    setSearchText: React.Dispatch<React.SetStateAction<string>>
+    searchApplied: boolean
+    setSearchApplied: React.Dispatch<React.SetStateAction<boolean>>
```
Comment (sohel-rp) [inline src/components/ApplicationGroup/SearchBar.tsx:18]: move to types

File: src/components/ApplicationGroup/Details/EnvCIDetails/EnvCIDetails.tsx [hunk 2]
```diff
@@ -24,58 +24,51 @@ export default function EnvCIDetails({ filteredApps }: AppGroupDetailDefaultType
...
-            getCIConfigList(envId).then((result) => {
-                setInitResult(result)
+            getCIConfigList(envId, filteredAppIds).then((result) => {
+                if (result?.pipelineList.length) {
+                    const _filteredPipelines = []
+                    let nonWebhookCIExist = false
+                    let selectedPipelineExist = false
+                    result.pipelineList.forEach((pipeline) => {
+                        _filteredPipelines.push(pipeline)
+                        nonWebhookCIExist = true
```
Comment (sohel-rp) [inline src/components/ApplicationGroup/Details/EnvCIDetails/EnvCIDetails.tsx:42]: is this expected? If yes then it'll always go in if branch on line 46

File: src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx [hunk 3]
```diff
@@ -183,6 +176,7 @@ export default function EnvCDDetails({ filteredApps }: AppGroupDetailDefaultType
                 />
             )
         } else {
+            const selectedApp = pipelineList.find((envData) => envData.appId === +appId)
```
Comment (sohel-rp) [inline src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx:179]: Can we use the same pattern either `+` or `Number`?

Comment (vivek-devtron) [reply src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx:179]: We can Use +

File: src/components/ApplicationGroup/CreateAppGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+        if (event.target.name === 'name') {
+            setAppGroupName(event.target.value)
+        } else {
+            setAppGroupDescription(event.target.value)
+        }
+    }
+
+    const handleFilterChanges = (_searchText: string): void => {
+        const _filteredData = appList.filter((app) => app.appName.indexOf(_searchText) >= 0)
+        //setFilteredAppList(_filteredData)
```
Comment (sohel-rp) [inline src/components/ApplicationGroup/CreateAppGroup.tsx:85]: remove if it's not be used

Comment (vivek-devtron) [reply src/components/ApplicationGroup/CreateAppGroup.tsx]: done

File: src/components/ApplicationGroup/Details/EnvCIDetails/EnvCIDetails.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/ApplicationGroup/Details/EnvCIDetails/EnvCIDetails.tsx:42]: Done

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**